Assets/Packages/com.studio23.ss2.upmassistant/Editor/Manager/PackageHandler.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Studio23.SS2.UPMAssistant.Editor.Data;
+using UnityEditor;
+using UnityEngine;
+
+namespace Studio23.SS2.UPMAssistant.Editor
+{
+    
+    public static class PackageHandler
+    {
+        public static string FilePath;
+        public static PackageJsonData LoadData()
+        {
+
+
+            if (File.Exists(FilePath))
+            {
+                var jsonString = File.ReadAllText(FilePath);
+                
+                if (string.IsNullOrWhiteSpace(jsonString))
+                {
+                    Debug.Log($"package.json is empty. Default data is loaded");
+                     return LoadDefaultPackageJsonData();
+                }
+                Debug.Log($"Previous data is loaded");
+               return  JsonConvert.DeserializeObject<PackageJsonData>(jsonString);
+            }
+
+            Debug.Log($"Default data loaded");
+            return  LoadDefaultPackageJsonData();
+        }
+
+        public static PackageJsonData CheckUpdateOnDisabledProperties(PackageJsonData jsonData)
+        {
+            var haveUpdate = UpdateProperties(jsonData);
+            if (haveUpdate)
+            {
+                SaveData(jsonData);
+            }
+
+            return jsonData;
+        }
+
+        public static void SaveData(PackageJsonData newJsonData)
+        {
+            string jsonString = JsonConvert.SerializeObject(newJsonData, Formatting.Indented);
+            JObject jsonObject = JObject.Parse(jsonString);
+
+            if (jsonObject == null || !jsonObject.HasValues)
+            {
+                Debug.LogError($"Data not found!");
+                return;
+            }
+
+            File.WriteAllText(FilePath, jsonString);
+            Debug.Log($"File created with data: {FilePath}");
+            AssetDatabase.Refresh();
+        }
+        
+        private static bool UpdateProperties(PackageJsonData jsonData)
+        {
+            var haveUpdate = false;
+
+            haveUpdate |= UpdateField(jsonData, "name", DataHandler.LoadPackageNameData());
+            haveUpdate |= UpdateField(jsonData, "version", Application.version);
+            haveUpdate |= UpdateField(jsonData, "displayName", Application.productName);
+            haveUpdate |= UpdateField(jsonData, "licensesUrl", DataHandler.LoadLicenseURLData());
+            haveUpdate |= UpdateField(jsonData.author, "name", Application.companyName);
+
+            return haveUpdate;
+        }
+        private static bool UpdateField<T>(object obj, string fieldName, T newValue)
```

Comment:
missing code comments

Reply:
// Check is there any update on unity PlayerSettings or GitHub license